federation/cmd/federation-controller-manager/app/options/options.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package options provides the flags used for the controller manager.
+
+package options
+
+import (
+	"time"
+
+	"github.com/spf13/pflag"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+	"k8s.io/kubernetes/pkg/client/leaderelection"
+)
+
+type ControllerManagerConfiguration struct {
+	unversioned.TypeMeta
+
+	// port is the port that the controller-manager's http service runs on.
+	Port int `json:"port"`
+	// address is the IP address to serve on (set to 0.0.0.0 for all interfaces).
+	Address string `json:"address"`
+	// clusterMonitorPeriod is the period for syncing ClusterStatus in cluster controller.
+	ClusterMonitorPeriod unversioned.Duration `json:"clusterMonitorPeriod"`
+	// APIServerQPS is the QPS to use while talking with federation apiserver.
+	APIServerQPS float32 `json:"federatedAPIQPS"`
+	// APIServerBurst is the burst to use while talking with federation apiserver.
+	APIServerBurst int `json:"federatedAPIBurst"`
+	// enableProfiling enables profiling via web interface host:port/debug/pprof/
+	EnableProfiling bool `json:"enableProfiling"`
+	// leaderElection defines the configuration of leader election client.
+	LeaderElection componentconfig.LeaderElectionConfiguration `json:"leaderElection"`
+}
+
+// CMServer is the main context object for the controller manager.
+type CMServer struct {
+	ControllerManagerConfiguration
+	Master          string
+	Kubeconfig      string
+}
+
+const (
+	// FederatedControllerManagerPort is the default port for the federation controller manager status server.
+	// May be overridden by a flag at startup.
+	FederatedControllerManagerPort = 10252
```

Comment:
This is what kube controller manager uses as well.
Will be good to use a different port here to prevent port conflict.
You can use 10253?